lib/pushmi_pullyu/swift_depositer.rb

```diff
@@ -0,0 +1,50 @@
+require 'openstack'
+
+class PushmiPullyu::SwiftDepositer
+
+  attr_reader :swift_connection, :swift_container
+
+  def initialize(connection, container)
+    raise 'conection can not be nil' if connection.nil?
+    raise 'container can not be nil' if container.nil?
+
+    endpoint     = connection[:endpoint]
```

Comment:
just plug endpoint/auth_version into the auth_url down on line 17. No need to create these variables